### DISCUSSION TITLE
Tenor V2?

### DISCUSSION THREAD

**Discussion by Mister-Hope** [walinejs/waline#1431] [Q&A] [answered]
@lizheming Why are we using v1 instead of v2?

https://developers.google.com/tenor/guides/quickstart

**Comment by lizheming** [comment, marked as answer]
The tenor v2 api host `https://tenor.googleapis.com` also forbidden for Chinese user. I'm not sure we should use V2.

**Reply by Mister-Hope** [reply]
> The tenor v2 api host `https://tenor.googleapis.com` also forbidden for Chinese user. I'm not sure we should use V2.

Tenor v1 is already unaccessible in most places. That's why I am hoping to find a plan to add more provider, since we already extract the logic with provided options .

**Reply by lizheming** [reply]
So how about use giphy as default to avoid the problem? https://github.com/orgs/walinejs/discussions/1433

**Reply by Mister-Hope** [reply]
I agree that, we can add tenor v1 and v2 config in snippet
